### PR TITLE
add aria-describedby to featured-image button

### LINKED ISSUE #14416: Describe the image in the featured image preview
When a featured image is set, a preview is displayed within a button. Pressing the preview button opens the media modal to replace the image.

The preview button has an `aria-label="Edit or update the image"` which gives the correct information to assistive technology users.

However, there's no indication of _what_ the set image is. All users need to know which image is set, including assistive technology users.

In this case, it's not possible to use the `alt` attribute because the entire content of the preview button is overridden by the `aria-label="Edit or update the image"`. A possible solution is an `aria-describedby` attribute on the button to target a visually hidden description.

The description should contain some meaningful text:
- when the image has an alt text: `Current image: altText`
- when the image doesn't have an alt text: `The current image has no alternative text. The file name is: fileName`

Note: the second text is the same currently used in core for the media widgets.

This way, the whole preview would be announced, for example, this way:
- Edit or update the image (brief pause) Current image: My awesome image
- Edit or update the image (brief pause) The current image has no alternative text. The file name is: DSC-345678.jpg

### PR DESCRIPTION
**Pull request by aristath**
Fixes #14416

## Description

This PR adds an `aria-describedby` attribute to the featured-image button. 
The `aria-describedby` attr will contain a reference to a hidden element that has the alt text, or if that doesn't exist it will use the filename.

## How has this been tested?

Tested by using featured images with and without alt text.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] ~~I've included developer documentation if appropriate.~~ 
- [ ] ~~I've updated all React Native files affected by any refactorings/renamings in this PR.~~

### CODE REVIEW

**Review by afercia — CHANGES_REQUESTED** [review]
Thanks for working on this @aristath !

Works well! From an a11y perspective, the only thing I'd suggest to change is to _not_ use the `media.slug` because it's not particularly relevant: the slug can be edited to something that is completely different from the file name. Also, the slug is not searchable in the media library while the file name is.

So for an image without alt text I'm getting this:

`The current image has no alternative text. The file name is: blue-beanie-jpg`

while it should use the file name instead:

`The current image has no alternative text. The file name is: blue-beanie.jpg`

Is it safe to assume there's always a `full` image size? If so, the file name to use is stored under `media.media_details.sizes['full'].file`.

File: packages/editor/src/components/post-featured-image/index.js [hunk 1]
```diff
@@ -95,6 +95,27 @@ function PostFeaturedImage( {
 		<PostFeaturedImageCheck>
 			{ noticeUI }
 			<div className="editor-post-featured-image">
+				{ media && (
+					<div
+						id={ `editor-post-featured-image-${ featuredImageId }-describedby` }
+						className={ 'hidden' }
```
Comment (afercia) [inline packages/editor/src/components/post-featured-image/index.js:101]: I think this doesn't need the brackets and can simply be `className="hidden"`

Comment (aristath) [reply packages/editor/src/components/post-featured-image/index.js]: Fixed :+1:

**Comment by aristath** [comment]
Thank you for the review @afercia 

> Is it safe to assume there's always a full image size? If so, the file name to use is stored under `media.media_details.sizes['full'].file`

Good point, I hadn't thought of that. You are correct, the `full` size will always exist. I switched it to use `media.media_details.sizes.full.file`, tested it and confirmed it works well :+1:

File: packages/editor/src/components/post-featured-image/index.js [hunk 2]
```diff
@@ -95,6 +95,27 @@ function PostFeaturedImage( {
...
+					>
+						{ media.alt_text &&
+							sprintf(
+								// Translators: %s: The selected image alt text.
+								__( 'Current image: %s' ),
+								media.alt_text
+							) }
+						{ ! media.alt_text &&
+							sprintf(
+								// Translators: %s: The selected image slug.
```
Comment (afercia) [inline packages/editor/src/components/post-featured-image/index.js:111]: This comment should not be updated, as it's not the slug any longer 🙂

Comment (aristath) [reply packages/editor/src/components/post-featured-image/index.js]: Fixed (changed `slug` to `filename`) :+1:

**Comment by afercia** [comment]
@aristath Looks good from an a11y perspective! Just two minor things that I'd totally leave up to you and the team:
- a translator comment should be updated
- not sure if it's best to check for `media.media_details.sizes.full.file` anyways, just to be 100% sure the full size exists and has the file info

**Review by afercia — APPROVED** [review]

**Comment by afercia** [comment]
You're fast! 😄

**Comment by aristath** [comment]
> not sure if it's best to check for media.media_details.sizes.full.file anyways, just to be 100% sure the full size exists and has the file info

I can't think of any reason why that wouldn't exist... But you're right, better safe than sorry. I added a fallback to the slug in case something goes terribly wrong and there is no full image-size or filename :+1:

**Pull request merged**